vms/platformvm/state/stakers_model_storage_test.go

```diff
@@ -0,0 +1,966 @@
+	},
+	DestroySystemUnderTestFunc: func(sut commands.SystemUnderTest) {
+		// retrieve base state and close it
+		sys := sut.(*sysUnderTest)
+		err := sys.baseState.Close()
+		if err != nil {
+			panic(err)
+		}
+	},
+	// a trick to force command regeneration at each sampling.
+	// gen.Const would not allow it
+	InitialStateGen: gen.IntRange(1, 2).Map(
+		func(int) *stakersStorageModel {
+			return newStakersStorageModel()
+		},
+	),
+
+	InitialPreConditionFunc: func(state commands.State) bool {
+		return true // nothing to do for now
+	},
+	GenCommandFunc: func(state commands.State) gopter.Gen {
+		return gen.OneGenOf(
+			genPutCurrentValidatorCommand,
+			genDeleteCurrentValidatorCommand,
+
+			genPutCurrentDelegatorCommand,
+			genDeleteCurrentDelegatorCommand,
+
+			genAddTopDiffCommand,
+			genApplyAndCommitBottomDiffCommand,
+			genRebuildStateCommand,
+		)
+	},
+}
+
+// PutCurrentValidator section
+type putCurrentValidatorCommand struct {
+	sTx *txs.Tx
+	err error
+}
+
+func (cmd *putCurrentValidatorCommand) Run(sut commands.SystemUnderTest) commands.Result {
+	sTx := cmd.sTx
+	sys := sut.(*sysUnderTest)
+
+	if err := sys.checkThereIsADiff(); err != nil {
+		return sys // state checks later on should spot missing validator
+	}
+
+	stakerTx := sTx.Unsigned.(txs.StakerTx)
+	currentVal, err := NewCurrentStaker(sTx.ID(), stakerTx, uint64(1000))
+	if err != nil {
+		return sys // state checks later on should spot missing validator
+	}
+
+	topChainState := sys.getTopChainState()
+	topChainState.PutCurrentValidator(currentVal)
+	topChainState.AddTx(sTx, status.Committed)
+	return sys
+}
+
+func (cmd *putCurrentValidatorCommand) NextState(cmdState commands.State) commands.State {
+	sTx := cmd.sTx
+	stakerTx := sTx.Unsigned.(txs.StakerTx)
+	currentVal, err := NewCurrentStaker(sTx.ID(), stakerTx, uint64(1000))
+	if err != nil {
+		return cmdState // state checks later on should spot missing validator
+	}
+
+	cmdState.(*stakersStorageModel).PutCurrentValidator(currentVal)
+	return cmdState
+}
+
+func (*putCurrentValidatorCommand) PreCondition(commands.State) bool {
+	// We allow inserting the same validator twice
+	return true
+}
+
+func (cmd *putCurrentValidatorCommand) PostCondition(cmdState commands.State, res commands.Result) *gopter.PropResult {
+	if cmd.err != nil {
+		cmd.err = nil // reset for next runs
+		return &gopter.PropResult{Status: gopter.PropFalse}
+	}
+
+	if !checkSystemAndModelContent(cmdState, res) {
+		return &gopter.PropResult{Status: gopter.PropFalse}
+	}
+
+	return &gopter.PropResult{Status: gopter.PropTrue}
+}
+
+func (cmd *putCurrentValidatorCommand) String() string {
+	stakerTx := cmd.sTx.Unsigned.(txs.StakerTx)
+	return fmt.Sprintf("\nputCurrentValidator(subnetID: %v, nodeID: %v, txID: %v, priority: %v, unixStartTime: %v, duration: %v)",
+		stakerTx.SubnetID(),
+		stakerTx.NodeID(),
+		cmd.sTx.TxID,
+		stakerTx.CurrentPriority(),
+		stakerTx.StartTime().Unix(),
+		stakerTx.EndTime().Sub(stakerTx.StartTime()),
+	)
```

Comment:
in case the framework finds a sequence of ops breaking invariants, it'll print it out (after trying to simplify it). String() prints relevent command information to ease up debugging.